### ISSUE TITLE
intellij_idea coding style can not format block comments to line comments at the end of a statement

### ISSUE THREAD

**Issue by pohlondrej** [pinterest/ktlint#2936]
## Expected Behavior
Ktlint successfully formats a block comment at the end of a statement into a line comment:

```kotlin
fun main() {
    println("test") // test
}
```

## Observed Behavior
Ktlint fails with:
```
14:36:33.661 [pool-1-thread-1] WARN com.pinterest.ktlint.rule.engine.internal.CodeFormatter -- Format was not able to resolve all violations which (theoretically) can be autocorrected in file Main.kt
```

### .editorconfig
```editorconfig
root = true

[*.{kt,kts}]
ktlint_code_style = intellij_idea
```
Note: This bug only occurs when `ktlint_code_style` is set to `intellij_idea`. When it's set to `ktlint_official`, ktlint formats the file correctly.

### Main.kt
```kotlin
fun main() {
    println("test") /* test */
}
```

## Environment
* Version of ktlint used: 1.4.*, 1.5.0 (1.3.1 has the same problem, but does not fail because of #2803 )
